### PR TITLE
Run insight pings only on user/org settings change

### PR DESCRIPTION
**Pull request by vovakulikov**
Fix insight ping and run event logger only if user/org settings have been changed

Based on @eseliger [comment](https://github.com/sourcegraph/sourcegraph/pull/21177#discussion_r636012627) from [this PR](https://github.com/sourcegraph/sourcegraph/pull/21177)

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
🌟

File: client/web/src/SourcegraphWebApp.tsx [hunk 1]
```diff
@@ -334,27 +334,17 @@ class ColdSourcegraphWebApp extends React.Component<SourcegraphWebAppProps, Sour
...
 
-        // Observe settings mutations for analytics
+        // Track static metrics fo code insights.
+        // Insight count, insights settings, observe settings mutations for analytics
         // Track add delete and update events of code insights via
-        // checking user/org settings on settings update
         this.subscriptions.add(
-            from(this.platformContext.settings)
+            combineLatest([from(this.platformContext.settings), authenticatedUser])
                 .pipe(bufferCount(2, 1))
```
Comment (felixfbecker) [inline client/web/src/SourcegraphWebApp.tsx:342]: Does this also fire on page load? Or only when settings were changed?

**Review by felixfbecker — APPROVED** [review]

File: client/web/src/SourcegraphWebApp.tsx (hunk 1 above)
Comment (vovakulikov) [reply client/web/src/SourcegraphWebApp.tsx:342]: @felixfbecker I checked this locally and this fires only when a user has changed something in user/org settings

Comment (vovakulikov) [reply client/web/src/SourcegraphWebApp.tsx:342]: I mean this fires on mount but we have logic within `logInsightMetrics` about preventing fire logs if a user hasn't changed settings so I believe we don't have the problem with firing logs on mount

**Pull request merged**